lasagne/layers/recurrent.py

```diff
@@ -0,0 +1,1239 @@
+"""
+Layers to construct recurrent networks. Recurrent layers can be used similarly
+to feed-forward layers except that the input shape is expected to be
+``(batch_size, sequence_length, num_inputs)``. The input is allowed to have
+more than three dimensions in which case dimensions trailing the third
+dimension are flattened.
```

Comment:
Hm, convolutional-input, convolutional-connection custom RNN layers is not something I've thought about.  For all the other (non-custom) layers, this is true.  Making a toy example/text like this just to verify that it works is a good idea.